src/lib/api/core/renderer-adapter.ts

```diff
@@ -0,0 +1,76 @@
+import {Renderer, Renderer2, RendererStyleFlags2} from '@angular/core';
+
+/**
+ * Adapts the 'deprecated' Angular Renderer v1 API to use the new Renderer2 instance
+ * This is required for older versions of NgStyle and NgClass: which require
+ * the v1 API (but should use the v2 instances)
+ */
+export class RendererAdapter extends Renderer {
```

Comment:
You are still referring to `Renderer`. I think you should drop that.

Reply:
@mhevery - To manually construct a NgClass I must pass it an instance of a class that `extends Renderer`. Or am I missing something ?

Reply:
Do you mean change `RendererAdapter` to `HybridAdapter` ?

Reply:
The whole point of the issue is that the `Renderer` symbol is being removed entirely.

Reply:
@jelbourn  -If we do this, then we break support for Angular 4.x

```js
// Create an instance NgClass Directive instance only if `ngClass=""` has NOT been defined on
    // the same host element; since the responsive variations may be defined...
    if ( !this._ngClassInstance ) {
      let adapter = new RendererAdapter(_renderer);
      this._ngClassInstance = new NgClass(_iterableDiffers, _keyValueDiffers, _ngEl, adapter);
    }
```

since NgClass in core expects the Renderer.

```js
import { Renderer } from '@angular/core';
export declare class NgClass implements DoCheck {
    private _iterableDiffers;
    private _keyValueDiffers;
    private _ngEl;
    private _renderer;
    private _iterableDiffer;
    private _keyValueDiffer;
    private _initialClasses;
    private _rawClass;
    constructor(_iterableDiffers: IterableDiffers, _keyValueDiffers: KeyValueDiffers, _ngEl: ElementRef, _renderer: Renderer);
```

Reply:
So I am not sure WHAT to do here because the darn `ngClass` extends an abstract base class instead of implements and interface.

Reply:
I could just copy the class implementation from ngClass (in core), but that seems to be a clumsy solution. Perhaps that is the only viable solution ?

Reply:
1) don't implement `Renderer`
2) Make sure that the Shape of the classes matches what `Renderer` expects (this should be noop since you implemented it.)
3) When passing it into `NgClass` just cast it to `any` so that the compiler is happy.